src/seedu/addressbook/commands/ClearCommand.java

```diff
@@ -10,13 +14,23 @@
             + "Example: " + COMMAND_WORD;
 
     public static final String MESSAGE_SUCCESS = "Address book has been cleared!";
+    public static final String MESSAGE_NO_CLEAR = "Address book has not been cleared.";
+
+    private final Scanner in = new Scanner(System.in);
+    private TextUi ui = new TextUi();
 
     public ClearCommand() {}
 
 
     @Override
     public CommandResult execute() {
-        addressBook.clear();
-        return new CommandResult(MESSAGE_SUCCESS);
+        System.out.print(ui.LINE_PREFIX + "Are you sure you want to clear the address book? (Y/N): ");
```

Comment:
Presently this clear command class is handling two roles: clearing the address book and textUI stuff. Does it violate SRP? 
If yes, try to think how you would fix that.